src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -340,6 +340,11 @@ private async Task<bool> UpdateManifestWithVersionAsync(string id, bool includeP
                     // Create version file later used as part of installing the workload set in the file-based installer and in the msi-based installer
                     using PackageArchiveReader packageReader = new(packagePath);
                     var downloadedPackageVersion = packageReader.NuspecReader.GetVersion();
+                    if (packageVersion != null && !downloadedPackageVersion.Equals(packageVersion))
+                    {
+                        throw new Exception(string.Format(Update.LocalizableStrings.WorkloadVersionNotFound, packageVersion));
```

Comment:
I think the exception message should specifically say that a different version was downloaded than we were expecting, and give both versions.

The exception message doesn't need to be localized in this case, as we think it's an internal error.